app/src/main/java/io/redgreen/benchpress/hellostranger/HelloStrangerActivity.kt

```diff
@@ -1,21 +1,86 @@
 package io.redgreen.benchpress.hellostranger
 
+import android.annotation.SuppressLint
 import android.content.Context
 import android.content.Intent
-import android.os.Bundle
-import android.support.v7.app.AppCompatActivity
-import io.redgreen.benchpress.R
+import com.spotify.mobius.Next
+import io.reactivex.ObservableTransformer
 import io.redgreen.benchpress.architecture.BaseActivity
+import kotlinx.android.synthetic.main.hello_stranger_activity.*
+import android.text.Editable
+import android.text.TextWatcher
+import android.widget.Toast
 
-class HelloStrangerActivity : AppCompatActivity() {
-  companion object {
-    fun start(context: Context) {
-      context.startActivity(Intent(context, HelloStrangerActivity::class.java))
+
+class HelloStrangerActivity :
+    BaseActivity<HelloStrangerModel, HelloStrangerEvent, HelloStrangerEffect>(), Interactor2 {
+    companion object {
+        fun start(context: Context) {
+            context.startActivity(Intent(context, HelloStrangerActivity::class.java))
+        }
+    }
+
+    override fun layoutResId(): Int {
+        return io.redgreen.benchpress.R.layout.hello_stranger_activity
+    }
+
+    override fun setup() {
+        nameEditText.addTextChangedListener(object : TextWatcher {
```

Comment:
create an open class which extends Textwatcher. Then pass that classes object here so that we dont have to override unused function.